src/identity.cc

```diff
@@ -772,13 +773,13 @@ TRITONBACKEND_ModelInstanceExecute(
         continue;
       }
 
-      std::string input_name_str = "INPUT" + std::string(output_name, 6, -1);
-      const char* input_name = input_name_str.c_str();
+      std::string index_str = std::string(output_name, 6, -1);
+      std::string input_name = "INPUT" + index_str;
       TRITONBACKEND_Input* input = nullptr;
-      if (input_names.find(input_name_str) != input_names.end()) {
+      if (input_indices.find(std::stoi(index_str)) != input_indices.end()) {
```

Comment:
So, basically finding by name in the set of strings was causing the slowdown? 
We need to check in our other backends if we are unintentionally comparing strings instead of indices. Would you mind looking at the same?

Reply:
Yes. The difference is in microseconds but definitely I will look into the other backends for similar fixes